cub/cub/detail/type_traits.cuh

```diff
@@ -66,5 +67,21 @@ using invoke_result_t =
 template <typename Invokable, typename InitT, typename InputT>
 using accumulator_t = typename ::cuda::std::decay<invoke_result_t<Invokable, InitT, InputT>>::type;
 
+/**********************************************************************************************************************
+ * Additional type traits
+ **********************************************************************************************************************/
+
+template<typename T>
+_CCCL_INLINE_VAR constexpr int num_bits_v = sizeof(T) * CHAR_BIT;
+
+template<typename T, typename... TArgs>
+_CCCL_INLINE_VAR constexpr bool are_same_v = ::cuda::std::conjunction_v<::cuda::std::is_same<T, TArgs>...>;
+
+template<typename T, typename... TArgs>
+_CCCL_INLINE_VAR constexpr bool is_one_of_v = ::cuda::std::disjunction_v<::cuda::std::is_same<T, TArgs>...>;
+
+template<typename...>
+_CCCL_INLINE_VAR constexpr bool always_false_v = false;
```

Comment:
We technically still compile against C++11 so this night need to be hardened against C++11